### PR TITLE
useImageColour Hook - colorthief

### PR DESCRIPTION
**Pull request by ryanmccombe**
Resolves #NUMBER

**Overall change:**
_A very high-level summary of easily-reproducible changes that can be understood by non-devs._

**Code changes:**

- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by DarioR01 — APPROVED** [review]
LGTM. I also find this very cool. Am I the only one 🤣 ?

**Comment by amoore108** [comment]
Looks great, fairly seemless transition to using colorthief by the looks of it 👍🏼

**Review by amoore108 — APPROVED** [review]

**Review by MuhammadAHussain — APPROVED** [review]
LGTM 👍

File: src/app/hooks/useImageColour/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+The first argument to the hook is a link to the image to be used.
+
+- For us to be able to download the image, the server must respond with appropriate CORS headers. BBC services, including iChef, do include the required headers.
+- Any image size beyond approximately 20 x 20 pixels is unnecessary. Unless a larger version of the image is being downloaded anyway for some other use, generally using a very small image will conserve bandwidth
+
+##### `options`
+
+The hook accepts a second, optional argument. This argument can contain various configuration options:
+
+- `contrastColour` - A hexidecimal colour string, eg `"#ff0005"`. If provided, this colour will be used in the assessment of contrast requirements, as described below```
```
Comment (rebeccamcginn) [inline src/app/hooks/useImageColour/README.md]:
```suggestion
- `contrastColour` - A hexidecimal colour string, eg `"#ff0005"`. If provided, this colour will be used in the assessment of contrast requirements, as described below.
```

File: src/app/hooks/useImageColour/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+- `rgb` - An array representing the colour's red, green and blue components, eg `[255, 0, 5]`
+- `isFallback` - A boolean representing whether this colour was the `fallbackColour` passed in the input options
+
+###### `isLoading`
+
+A boolean that will be true if the hook is currently analysing an image. Note that changing the URL passed to the hook when this value is `true` will have no effect.
+
+###### `error`
+
+Will be null if no error has occured, or an error string if something has gone wrong. See the section on Error Handling below.
```
Comment (rebeccamcginn) [inline src/app/hooks/useImageColour/README.md]:
```suggestion
Will be null if no error has occurred, or an error string if something has gone wrong. See the section on Error Handling below.
```

**Pull request merged**